src/pybind/mgr/dashboard/frontend/src/app/shared/directives/form-input-disable.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { FormNameDirective } from './form-name.directive';
+
+@Directive({
+  selector:
+    'input:not([cdNoFormInputDisable]), select:not([cdNoFormInputDisable]), [cdFormInputDisable]'
+})
+export class FormInputDisableDirective implements AfterViewInit {
+  permissions: Permissions;
+
+  @Input() disabled: boolean;
+
+  constructor(
+    @Optional() private service: FormNameDirective,
```

Comment:
It is still not possible to use this in a single form input.
The permission scope always comes from the form element and is applied to all inputs.

Reply:
Yes. I think I can try and come up with a followup for this with the possibility of single form input disabling based on the scope. @tspmelo